src/components/Navbar/NavLanguageSelector.tsx

```diff
@@ -40,15 +52,35 @@ export default function NavLanguageSelector({
     window.location.reload()
   }
 
+  const desktopDropdownStyle: CSSProperties = {
+    border: '1px solid ' + colors.stroke.tertiary,
+    marginRight: 20,
+    boxShadow: juiceBoxShadow(isDarkMode),
+  }
+
+  const selectHeader = mobile
+    ? Languages[currentSelectedLanguage].long
+    : Languages[currentSelectedLanguage].short
+  // Close dropdown when clicking anywhere in the window
+  window.addEventListener('click', () => setDropdownOpen(false), false)
   return (
-    <div className="language-selector">
-      <GlobalOutlined />
+    <div
+      className="language-selector"
+      style={{ cursor: 'pointer' }}
+      onClick={e => {
+        e.stopPropagation()
+        setDropdownOpen(dropdownOpen === true ? false : true)
+      }}
+    >
+      <GlobalOutlined style={{ marginBottom: 2 }} />
       <Select
         className="medium"
         style={{
           ...selectStyle,
         }}
-        value={Languages[currentSelectedLanguage]?.long ?? 'English'}
+        dropdownStyle={mobile ? {} : desktopDropdownStyle}
+        open={dropdownOpen}
+        value={selectHeader ?? 'EN'}
```

Comment:
**suggestion** might be possible to source the default language from `i18n.ts`, not sure.